### PR TITLE
[REVIEW] Remove deprecated CUDA library calls

### PR DESCRIPTION
**Pull request by aschaffer**
This tracks work on https://github.com/rapidsai/cugraph/issues/869.

### CODE REVIEW

File: cpp/src/utilities/spmv_1D.cu [hunk 1]
```diff
@@ -1,72 +1,86 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <raft/spectral/matrix_wrappers.hpp>
+#include "spmv_1D.cuh"
+
+namespace cugraph {
+namespace opg {
+template <typename vertex_t, typename edge_t, typename weight_t>
+OPGcsrmv<vertex_t, edge_t, weight_t>::OPGcsrmv(raft::handle_t const &handle_,
```
Comment (BradReesWork) [inline cpp/src/utilities/spmv_1D.cu:22]: thought we replaced OPGcsrmv with MGcsrmv

File: cpp/src/utilities/spmv_1D.cu [hunk 2]
```diff
@@ -1,72 +1,86 @@
...
+  v_glob = part_off[p - 1] + local_vertices[p - 1];
+  v_loc  = local_vertices[i];
+  vertex_t tmp;
+  CUDA_TRY(cudaMemcpy(&tmp, &off[v_loc], sizeof(vertex_t), cudaMemcpyDeviceToHost));
+  e_loc = tmp;
+  y_loc.resize(v_loc);
+}
+
+template <typename vertex_t, typename edge_t, typename weight_t>
+OPGcsrmv<vertex_t, edge_t, weight_t>::~OPGcsrmv()
```
Comment (BradReesWork) [inline cpp/src/utilities/spmv_1D.cu:49]: OPG should be MG

File: cpp/src/utilities/spmv_1D.cuh [hunk 3]
```diff
@@ -17,50 +17,44 @@
...
 
  public:
-  OPGcsrmv(const raft::comms::comms_t& comm,
-           VT* local_vertices,
-           VT* part_off,
-           ET* off_,
-           VT* ind_,
-           WT* val_,
-           WT* x);
+  OPGcsrmv(raft::handle_t const& handle_,
```
Comment (BradReesWork) [inline cpp/src/utilities/spmv_1D.cuh:46]: OPG should be MG

File: cpp/src/utilities/spmv_1D.cu (hunk 1 above)
Comment (seunghwak) [inline cpp/src/utilities/spmv_1D.cu:22]: Renaming OPGcsrmv to MGcsrmv will require too much work? Or isn't this something we can do in short time?

File: cpp/src/utilities/spmv_1D.cu [hunk 4]
```diff
@@ -1,72 +1,86 @@
...
+ * limitations under the License.
+ */
+#include <raft/spectral/matrix_wrappers.hpp>
+#include "spmv_1D.cuh"
+
+namespace cugraph {
+namespace opg {
+template <typename vertex_t, typename edge_t, typename weight_t>
+OPGcsrmv<vertex_t, edge_t, weight_t>::OPGcsrmv(raft::handle_t const &handle_,
+                                               const raft::comms::comms_t &comm_,
```
Comment (seunghwak) [inline cpp/src/utilities/spmv_1D.cu:23]: Handle includes a communicator, why pass handle & comm separately?

File: cpp/src/utilities/spmv_1D.cu [hunk 5]
```diff
@@ -1,72 +1,86 @@
...
+                                               vertex_t *local_vertices_,
```
Comment (seunghwak) [inline cpp/src/utilities/spmv_1D.cu:24]: Variable names with trailing _ is for class private member variables (RAFT clang-tidy rules). I think we'd better follow this too avoid additional future changes.

File: cpp/src/utilities/spmv_1D.cuh [hunk 6]
```diff
@@ -17,50 +17,44 @@
...
+template <typename vertex_t, typename edge_t, typename weight_t>
 class OPGcsrmv {
  private:
   size_t v_glob;
   size_t v_loc;
   size_t e_loc;
-  const raft::comms::comms_t& comm;
-  VT* part_off;
-  VT* local_vertices;
+  raft::comms::comms_t const& comm;
```
Comment (seunghwak) [inline cpp/src/utilities/spmv_1D.cuh:31]: We're storing a reference to the handle in line 43 (https://github.com/rapidsai/cugraph/pull/1019/files#diff-da69c92a20221e39f984232cbe12b095R43). As the handle has a pointer to the communicator, should we store references to both handle and communicator?

File: cpp/src/utilities/spmv_1D.cu (hunk 1 above)
Comment (BradReesWork) [reply cpp/src/utilities/spmv_1D.cu:22]: I am fine adding that to a refactoring task in 0.16

File: cpp/src/utilities/spmv_1D.cu (hunk 5 above)
Comment (BradReesWork) [reply cpp/src/utilities/spmv_1D.cu:24]: +1

File: cpp/src/utilities/spmv_1D.cu (hunk 1 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cu]: Renaming done.

File: cpp/src/utilities/spmv_1D.cuh (hunk 3 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cuh]: Done.

File: cpp/src/utilities/spmv_1D.cu (hunk 2 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cu]: Done.

File: cpp/src/utilities/spmv_1D.cu (hunk 1 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cu]: Done.

File: cpp/src/utilities/spmv_1D.cu (hunk 5 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cu]: Fixed.

File: cpp/src/utilities/spmv_1D.cuh (hunk 6 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cuh]: Fixed.

File: cpp/src/utilities/spmv_1D.cu (hunk 4 above)
Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cu]: Fixed.

File: cpp/src/utilities/spmv_1D.cu [hunk 7]
```diff
@@ -1,72 +1,86 @@
...
+  auto stream = handle_.get_stream();
+
+  auto const &comm{handle_.get_comms()};
+
+  std::vector<size_t> recvbuf(comm.get_size());
+  std::copy(local_vertices_, local_vertices_ + comm.get_size(), recvbuf.begin());
+  comm.allgatherv(y_loc_.data().get(), x, recvbuf.data(), part_off_, stream);
+}
+
+template class MGcsrmv<int, int, double>;
```
Comment (ChuckHastings) [inline cpp/src/utilities/spmv_1D.cu:82]: I'd prefer int32_t, int32_t, {double/float} personally.  We might later want to support int64_t and it keeps things a bit clearer for the reader.

Comment (aschaffer) [reply cpp/src/utilities/spmv_1D.cu]: Done.

**Comment by aschaffer** [comment]
rerun tests.

**Comment by BradReesWork** [comment]
rerun tests

**Pull request merged**